ui/prometheus-plugin/src/plugins/prometheus-variables.ts

```diff
@@ -44,6 +46,15 @@ function getQueryOptions(ctx: GetVariableOptionsContext) {
   return queryOptions;
 }
 
+function interpolatePluginDefinition(def: object, ctx: GetVariableOptionsContext) {
+  return replaceTemplateVariablesInObject(def, ctx.variables);
+}
+
+const dependsOn: VariablePlugin['dependsOn'] = (definition) => {
+  const stringDefinition = JSON.stringify(definition.spec.plugin.spec);
```

Comment:
A little worried about using `JSON.stringify` here (and above in the `replace` function). Is there a way to write these without needing it? Just worried about the cost of it potentially being called frequently. I think I've just seen too many projects where JSON stringify/parse usage ended up being a performance issue later. 😂

Reply:
Haha yeah that's a fair argument. I will rewrite it so it looks at the specific properties that may have variables. I mostly did JSON.stringify as a quick way to iterate since the size of a definition would be relatively small vs a dynamic payload of objects.